File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/non-plagiarized/05/T07.py
import sys

def sum_major_diagonal(matrix):
    diagonal_sum = 0
    for i in range(4):
        diagonal_sum += matrix[i][i]
    return diagonal_sum

if __name__ == "__main__":
    matrix = []
    print("Enter a 4-by-4 matrix row by row:")
    for _ in range(4):
        row = [float(x) for x in input().split()]
        matrix.append(row)
    
    print(f"Sum of the elements in the major diagonal is {sum_major_diagonal(matrix)}")

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/non-plagiarized/05/T07.ts
import * as readline from 'readline';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function sumMajorDiagonal(m: number[][]): number {
  let jumdiagonal = 0;
  for (let i = 0; i < 4; i++) {
    for (let j = 0; j < 4; j++) {
      if (i === j) {
        jumdiagonal += m[i][j];
      }
    }
  }
  return jumdiagonal;
}

rl.question('Enter a 4-by4 matrix row by row:\n', (input) => {
  const numbers: number[][] = [];
  const rows = input.trim().split('\n');
  for (const row of rows) {
    const values = row.split(/\s+/).map(Number);
    numbers.push(values);
  }

  console.log(`Sum of the elements in the major diagonal is ${sumMajorDiagonal(numbers)}`);
  rl.close();
});

